neuvector-scanner.advisories.yaml

```diff
@@ -20,3 +20,8 @@ advisories:
             componentType: go-module
             componentLocation: /usr/bin/neuvector-scanner
             scanner: grype
+      - timestamp: 2024-04-23T16:21:15Z
+        type: false-positive-determination
+        data:
+          type: vulnerable-code-not-included-in-package
+          note: The current version of the module v0.0.0-20240411013726-6a6b32c3e597 does not include the vulnerable code
```

Comment:
Why not? Is this more of a version misidentification issue? Could we demonstrate using the git history of `github.com/neuvector/neuvector` that the fix to this CVE is an ancestor of the current version?

Reply:
The version used in the package correspond to the latest available code (so it lives in main and it will be part of the next release). The commit was included as part of this PR: https://github.com/neuvector/neuvector/pull/1295

You can find a portion of the commit in the logs of main:
```
commit 19e43f8100a9d53bda5d02c62a877c11bb698f7e
Merge: 6ced20f8 6a6b32c3
Author: becitsthere <55168550+becitsthere@users.noreply.github.com>
Date:   Thu Apr 11 10:04:07 2024 -0700
    Merge pull request #1295 from becitsthere/main

    Cherry-pick scanner cache related commits
 ```
 
With that said, that commit of neuvector contains the fixed code mentioned in the **GHSA** and release [notes](https://open-docs.neuvector.com/releasenotes/5x/).

https://github.com/neuvector/neuvector/security/advisories/GHSA-622h-h2p8-743x

Reply:
Do we know where in the git history of `github.com/neuvector/neuvector` at `v0.0.0-20240411013726-6a6b32c3e597` the fix was applied?

I see you mentioned `19e43f8100a9d53bda5d02c62a877c11bb698f7e` but I don't see that in the git history there. Let's convince ourselves that we've found where the fix (or even just fixed release) of the module occurred, and then we can test that that commit is an ancestor of `6a6b32c3e597`, and then we're golden. But we should be able to say when (some git commit) we think the fix occurred, to prove we understand what's happening here.

And if that's the case, we can use `vulnerable-code-version-not-used`, with our explanation of the above.

Reply:
The commit version being used has the CVE mitigated. But the project itself doesn't have a release -tag and the version tagged for the package is not some github version. So although there is go/bump applied the scanner doesn't pick the updated go.mod

Reply:
> The commit version being used has the CVE mitigated

...As of which commit? <- that's what I'm curious to know

Reply:
Sorry I mean in the vulnerable repo — `github.com/neuvector/neuvector`

Reply:
This [commit](https://github.com/neuvector/neuvector/commit/be746957ee7c4e343b4e2ed261f767a43c184327) in the repo — github.com/neuvector/neuvector  seems to have it fixed. 

https://github.com/neuvector/neuvector/releases?page=2#:~:text=Compare-,5.2.2%20Release,-Security%20Advisory%20for

Reply:
Okay gotcha. It looks like that commit (`be746957`) isn't in the history of the commit used by neuvector/scanner (`6a6b32c3`). That would've made this a cleaner explanation — but that's okay. It looks like the merge-base of v5.2.2 and `6a6b32c3` is `ef750ee9`, which covers most of the changes in the v5.2.2 release, so it's probably a safe bet.

If we end up finding the exact commit in neuvector/neuvector with the fix (i.e. not just the release), and can prove that that commit is an ancestor of `6a6b32c3`, that's ideal. But if not, we can say vulnerable version not used and say something to the effect of neuvector scanner using a version after the fix.

Reply:
Sorry, I didn't have time to come back to this PR yet. Yes, one of the fixes for the CVE got merged from this PR https://github.com/neuvector/neuvector/pull/1030 which matches the commit id that you mentioned `ef750ee9`. I also realized the commit `19e43f` was pointing to a totally wrong PR (which explains our initial issue).

If we run `git branch --contains ef750ee9` on the neuvector/neuvector branch used by neuvector/scanner, it shows that it contains that commit. I also tried `git merge-base --is-ancestor ef750ee9 6a6b32c` so it shows it as an ancestor.

Reply:
Yep! Great. So once we get the `type` and the `note` updated, this looks GTG